sqlmesh/utils/transactional_file.py

```diff
@@ -102,27 +93,28 @@ def acquire_lock(self, blocking: bool = True, timeout: float = -1) -> bool:
         not (for example if the timeout expired).
         """
         if self._is_locked:
-            # Make this idempotent for a single instance.
             return True
-        wait_time = 0.0
+
         self._fs.touch(self.lock_path)
         locks = self._sync_locks()
         active_lock = min(locks, key=locks.get)  # type: ignore
-        while active_lock != os.path.basename(self.lock_path):
-            if not blocking:
-                self._fs.rm(self.lock_path)
-                return False
-            if timeout > 0 and wait_time > timeout:
+        start = time.time()
+
+        while active_lock != self.lock_path:
+            if not blocking or (timeout > 0 and time.time() - start > timeout):
                 self._fs.rm(self.lock_path)
                 return False
-            time.sleep(POLLING_INTERVAL)
-            wait_time += POLLING_INTERVAL
+
+            time.sleep(random.random() + POLLING_INTERVAL)
             locks = self._sync_locks()
-            if not locks:
-                break
+            if self.lock_path not in locks:
```

Comment:
So theoretically if the timeout is set to -1 it's still bounded by the lock file TTL. This is a non-obvious behavior.

Reply:
should i recreate the lockfile in this case?

Reply:
I believe we should recreate the file instead or do something to it to update its mtime